lib/azure/armrest/resource_group_based_service.rb

```diff
@@ -190,6 +195,13 @@ def delete(name, rgroup = configuration.resource_group)
 
       private
 
+      # Determines whether or not the calling method will perform a deep
+      # transform plus camelize on any hash options.
+      #
+      def transform_keys?
```

Comment:
For template deployment the `options` has structure [here](https://github.com/ManageIQ/manageiq-providers-azure/blob/master/app/models/manageiq/providers/azure/cloud_manager/orchestration_stack.rb#L21)

```
  { 'properties' => {
    :template => ...
    :parameters => ...
    :other...
}
```
Therefore when override `transform_create_options` we can skip `properties.parameters` and continue to transform others keys.

Reply:
@bzwei shouldn't we skip the template content as well?

Reply:
The template should be a string. But yes, skip it for safety.